examples/dampened_oscillator/dampened_oscillator_declarative.py

```diff
@@ -0,0 +1,133 @@
+from numerous.multiphysics.equation_decorators import Equation
+from numerous.multiphysics.equation_base import EquationBase
+from numerous.engine.system.item import Item
+from numerous.engine.system import Subsystem, ConnectorTwoWay, ItemsStructure
+from numerous.declarative.specification import ScopeSpec, ItemsSpec, Module, EquationSpec
+from numerous.declarative.variables import Parameter, Constant, State
+from numerous.declarative.bus import Connector, create_connections, get_value_for, set_value_from
+
+import numpy as np
+
+class DampenedOscillator(Module):
+    """
+        Equation and item modelling a spring and dampener
+    """
+    tag = "dampened_oscillator"
+
+    class Mechanics(ScopeSpec):
+        # Define variables
+        k = Constant(1) # spring constant
+        c = Constant(1) # coefficient of friction
+        a = Parameter(0) # acceleration
+        x, x_dot = State(0) # distance
+        v, v_dot = State(0) # velocity
+
+    mechanics = Mechanics()
+
+    coupling = Connector(
+        x = set_value_from(mechanics.x),
+        F = get_value_for(mechanics.v_dot)
+    )
+
+    @EquationSpec(mechanics)
+    def eval(self, scope: Mechanics):
+        scope.a = -scope.k * scope.x - scope.c * scope.v
+        scope.v_dot = scope.a
+        scope.x_dot = scope.v
+
+
+class SpringCoupling(Module):
+
+    tag = "springcoup"
+
+    class Mechanics(ScopeSpec):
+        k = Parameter(1)
+        c = Parameter(1)
+        F1 = Parameter(0)
+        F2 = Parameter(0)
+        x1 = Parameter(0)
+        x2 = Parameter(0)
+
+    mechanics = Mechanics()
+
+    class Items(ItemsSpec):
+        side1: DampenedOscillator
+        side2: DampenedOscillator
+
+    side1 = Connector(
+        F = set_value_from(mechanics.F1),
+        x = get_value_for(mechanics.x1)
+    )
+
+    side2 = Connector(
+        F=set_value_from(mechanics.F2),
+        x=get_value_for(mechanics.x2)
+    )
+
+    items = Items()
+
+    def __init__(self, tag="springcoup", k=1.0):
+        super().__init__(tag)
+
+        self.mechanics.set_values(k=k)
+
+    @EquationSpec(mechanics)
```

Comment:
Why do you need @EquationSpec here and not @Equation ?

Reply:
The equation spec allows passing of the instance of the ScopeSpec which internally adds the equation to the namespace mechanics. In this way the previous code of add_to_namespace can be omitted, while it is still explicit that the equation is connected to the namespace.